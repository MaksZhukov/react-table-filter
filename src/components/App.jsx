import React from 'react';
import GridLayout from 'react-grid-layout';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import PanelContainer from '../containers/Panel';

const App = ({ panels, addPanel }) => (
  <>
    <button type="button" className="btn" onClick={addPanel}>
      +
    </button>
    <GridLayout className="panels" cols={4} rowHeight={300} width={1600}>
      {panels.map((panel, key) => {
        const y = Math.floor(key / 4);
        const x = key % 4;
        return (
          <div
            key={panel.get('id')}
            data-grid={{
              x, y, w: 1, h: 1,
            }}
          >
            <PanelContainer x={x} y={y} id={panel.get('id')} />
          </div>
        );
      })}
    </GridLayout>
  </>
);

App.propTypes = {
  panels: PropTypes.instanceOf(List),
  addPanel: PropTypes.func,
};

export default App;
