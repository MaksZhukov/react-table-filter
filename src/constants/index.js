const SUB_ACTIONS_TYPES = ['SUCCESS', 'ERROR', 'PENDING'];
const TYPE_SEARCH = {
  all: '**',
  startWith: '*_',
  exact: '""',
};
const ORDER_CELLS = {
  up: 'A-Z',
  down: 'Z-A',
};
const OFFSETS_FILTERS = {
  height: 300,
  width: 400,
  indentX: 110,
  indentY: 100,
  additionalIndentY: 10,

};
export {
  SUB_ACTIONS_TYPES,
  TYPE_SEARCH,
  ORDER_CELLS,
  OFFSETS_FILTERS,
};
