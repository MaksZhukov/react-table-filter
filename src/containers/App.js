import { connect } from 'react-redux';
import App from '../components/App';
import { addPanel } from '../actions';
import { panelsSelector } from '../selectors';

const mapStateToProps = state => ({ panels: panelsSelector(state) });

const mapDispatchToProps = dispatch => ({ addPanel: () => dispatch(addPanel()) });

export default connect(mapStateToProps, mapDispatchToProps)(App);
