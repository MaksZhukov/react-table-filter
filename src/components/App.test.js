import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import { List, fromJS } from 'immutable';
import { ORDER_CELLS, TYPE_SEARCH } from '../constants';
import App from './App';


configure({ adapter: new Adapter() });
describe('component App', () => {
  const props = {
    panels: List([]),
    addPanel: jest.fn(),
  };
  it('render component App without crashing', () => {
    const wrapper = shallow(<App {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('render component App has btn for add panels', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper.find('.btn').length).toBe(1);
  });
  it('render component App without crashing and change props with new render', () => {
    const wrapper = shallow(<App {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({
      panels: List([fromJS({
        id: 10,
        isOpenFilters: false,
        filters: {
          defaultContexts: [],
          contexts: [],
          defaultDimensions: [],
          dimensions: [],
          cells: [],
          checkedAllCells: false,
          orderCells: ORDER_CELLS.up,
          search: {
            value: '',
            type: TYPE_SEARCH.all,
          },
        },
        responseGetTables: {},
      })]),
    });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('render component App has btn for add panels', () => {
    const wrapper = shallow(<App {...props} />);
    expect(wrapper.find('.btn').length).toBe(1);
  });
  it('render component App should call mock function when button is clicked', () => {
    const wrapper = shallow(<App {...props} />);
    wrapper.find('button').simulate('click');
    expect(props.addPanel).toHaveBeenCalled();
  });
});
