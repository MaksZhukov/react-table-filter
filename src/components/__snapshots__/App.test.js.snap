// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component App render component App without crashing 1`] = `
<Fragment>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <ReactGridLayout
    autoSize={true}
    className="panels"
    cols={4}
    compactType="vertical"
    containerPadding={null}
    draggableCancel=""
    draggableHandle=""
    isDraggable={true}
    isResizable={true}
    layout={Array []}
    margin={
      Array [
        10,
        10,
      ]
    }
    maxRows={Infinity}
    onDrag={[Function]}
    onDragStart={[Function]}
    onDragStop={[Function]}
    onLayoutChange={[Function]}
    onResize={[Function]}
    onResizeStart={[Function]}
    onResizeStop={[Function]}
    preventCollision={false}
    rowHeight={300}
    style={Object {}}
    useCSSTransforms={true}
    verticalCompact={true}
    width={1600}
  />
</Fragment>
`;

exports[`component App render component App without crashing and change props with new render 1`] = `
<Fragment>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <ReactGridLayout
    autoSize={true}
    className="panels"
    cols={4}
    compactType="vertical"
    containerPadding={null}
    draggableCancel=""
    draggableHandle=""
    isDraggable={true}
    isResizable={true}
    layout={Array []}
    margin={
      Array [
        10,
        10,
      ]
    }
    maxRows={Infinity}
    onDrag={[Function]}
    onDragStart={[Function]}
    onDragStop={[Function]}
    onLayoutChange={[Function]}
    onResize={[Function]}
    onResizeStart={[Function]}
    onResizeStop={[Function]}
    preventCollision={false}
    rowHeight={300}
    style={Object {}}
    useCSSTransforms={true}
    verticalCompact={true}
    width={1600}
  />
</Fragment>
`;

exports[`component App render component App without crashing and change props with new render 2`] = `
<Fragment>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <ReactGridLayout
    autoSize={true}
    className="panels"
    cols={4}
    compactType="vertical"
    containerPadding={null}
    draggableCancel=""
    draggableHandle=""
    isDraggable={true}
    isResizable={true}
    layout={Array []}
    margin={
      Array [
        10,
        10,
      ]
    }
    maxRows={Infinity}
    onDrag={[Function]}
    onDragStart={[Function]}
    onDragStop={[Function]}
    onLayoutChange={[Function]}
    onResize={[Function]}
    onResizeStart={[Function]}
    onResizeStop={[Function]}
    preventCollision={false}
    rowHeight={300}
    style={Object {}}
    useCSSTransforms={true}
    verticalCompact={true}
    width={1600}
  >
    <div
      data-grid={
        Object {
          "h": 1,
          "w": 1,
          "x": 0,
          "y": 0,
        }
      }
      key="10"
    >
      <Connect(Panel)
        id={10}
        x={0}
        y={0}
      />
    </div>
  </ReactGridLayout>
</Fragment>
`;
